File: app/src/main/java/com/gmmp/easylearn/activity/NavegacaoActivity.java
package com.gmmp.easylearn.activity;

import android.os.Bundle;
import android.support.design.widget.BottomNavigationView;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;
import android.support.annotation.NonNull;
import android.view.MenuItem;
import android.widget.TextView;

import com.gmmp.easylearn.R;
import com.gmmp.easylearn.fragment.DestaquesFragment;
import com.gmmp.easylearn.fragment.DisciplinasFragment;
import com.gmmp.easylearn.fragment.MensagensFragment;
import com.gmmp.easylearn.fragment.MeusCursosFragment;
import com.gmmp.easylearn.fragment.MinhaContaFragment;

public class NavegacaoActivity extends AppCompatActivity {

    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {

            switch (item.getItemId()) {
                case R.id.navigation_destaques:
                    mudarFragment(new DestaquesFragment());
                    return true;
                case R.id.navigation_meus_cursos:
                    mudarFragment(new MeusCursosFragment());
                    return true;
                case R.id.navigation_disciplinas:
                    mudarFragment(new DisciplinasFragment());
                    return true;
                case R.id.navigation_mensagens:
                    mudarFragment(new MensagensFragment());
                    return true;
                case R.id.navigation_minha_conta:
                    mudarFragment(new MinhaContaFragment());
                    return true;
            }

            return false;
        }
    };

    /**
     * Substitui o Fragment atual pelo fragment enviado como parêmetro.
     * @param fragment
     */
    private void mudarFragment(Fragment fragment) {
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.frameContainer, fragment);
        transaction.commit();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_navegacao);
        BottomNavigationView navView = findViewById(R.id.nav_view);
        navView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
    }

}

File: app/src/main/java/com/gmmp/easylearn/activity/NavegacaoActivity.kt
package com.gmmp.easylearn.activity

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.gmmp.easylearn.R
import com.gmmp.easylearn.fragment.*
import com.gmmp.easylearn.model.Curso
import com.gmmp.easylearn.model.Modulo
import com.gmmp.easylearn.model.Usuario
import com.gmmp.easylearn.model.Video
import com.google.android.material.bottomnavigation.BottomNavigationView

class NavegacaoActivity : AppCompatActivity() {

    // Isso aqui evita o problema com as variáveis globais
    companion object {
        var cursoGlobal = Curso()
        var moduloGlobal = Modulo()
        var videoGlobal = Video()
        var usuarioGlobal = Usuario()
    }

    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.navigation_destaques -> {
                mudarFragment(DestaquesFragment())
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_meus_cursos -> {
                mudarFragment(MeusCursosFragment())
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_disciplinas -> {
                mudarFragment(DisciplinasFragment())
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_mensagens -> {
                mudarFragment(MeuCanalFragment())
                return@OnNavigationItemSelectedListener true
            }
            R.id.navigation_minha_conta -> {
                mudarFragment(PreferenciasFragment())
                return@OnNavigationItemSelectedListener true
            }
        }

        false
    }

    /**
     * Substitui o Fragment atual pelo fragment enviado como parêmetro.
     * @param fragment
     */
    private fun mudarFragment(fragment: Fragment) {
        val transaction = supportFragmentManager.beginTransaction()
        transaction.replace(R.id.frameContainer, fragment)
        transaction.commit()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_navegacao)
        val navView = findViewById<BottomNavigationView>(R.id.nav_view)
        navView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)

        inicializar()
    }

    private fun inicializar() {
        //Esconde a actionBar
        supportActionBar?.hide()

        // Inicia o primeiro fragment para exibição.
        val transaction = supportFragmentManager.beginTransaction()
        transaction.add(R.id.frameContainer, DestaquesFragment())
        transaction.commit()
    }

}
